Lecture4/src/test/java/CalcUnitTests.java

```diff
@@ -0,0 +1,115 @@
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class CalcUnitTests {
+
+	@BeforeClass
+	public void oneTimeSetUp() {
+		System.out.print("Before Class");
+	}
+
+        @Test(dataProvider = "testAddDataProvider")
+        public void checkSum(double operand1, double operand2, double total) {
+            Calculator calculator = new Calculator(operand1, Operator.PLUS, operand2);
+            System.out.println("Test Total: " + calculator.makeCalculation());
+            Assert.assertEquals(calculator.makeCalculation(), total);
+        }
+
+	@DataProvider(name = "testAddDataProvider")
+	public Object[][] parameterAddTestProvider() {
+		return new Object[][]{
+				{2, 2, 4},
+				{1, 0, 1},
+				{0, 0, 0},
+				{1, -1, 0},
+				{-2, -2, -4},
+				{2, 22, 24},
+				{Double.MAX_VALUE, 1, Double.MAX_VALUE}
+		};
+	}
+
+        @Test(dataProvider = "testSubDataProvider")
+        public void checkDifference(double operand1, double operand2, double total) {
+            Calculator calculator = new Calculator(operand1, Operator.MINUS, operand2);
+            System.out.println("Test Total: " + calculator.makeCalculation());
+            Assert.assertEquals(calculator.makeCalculation(), total);
+        }
+
+	@DataProvider(name = "testSubDataProvider")
+	public Object[][] parameterSubTestProvider() {
+		return new Object[][]{
+				{2, 2, 0},
+				{1, 0, 1},
+				{0, 0, 0},
+				{0, 1, -1},
+				{-2, 2, -4},
+				{-2, 22, -24},
+				{Double.MIN_VALUE, 1, -1}
+		};
+	}
+
+        @Test(dataProvider = "testMultDataProvider")
+        public void checkProduct(double operand1, double operand2, double total) {
+            Calculator calculator = new Calculator(operand1, Operator.MULTIPLY, operand2);
+            System.out.println("Test Total: " + calculator.makeCalculation());
+            Assert.assertEquals(calculator.makeCalculation(), total);
+        }
+
+	@DataProvider(name = "testMultDataProvider")
+	public Object[][] parameterMultTestProvider() {
+		return new Object[][]{
+				{2, 2, 4},
+				{1, 0, 0},
+				{0, 0, 0},
+				{0, -1, -0.0},
+				{2, -2, -4},
+				{2, 22, 44}
+		};
+	}
+
+        @Test(expectedExceptions = IllegalArgumentException.class,
+                expectedExceptionsMessageRegExp = "Error: Can't divide by zero!",
+                dataProvider = "testDivByZeroDataProvider")
+        public void checkDivisionByZero(double operand1, double operand2, double total)
+                throws IllegalArgumentException {
```

Comment:
this method should throw `IllegalArgumentException`

Reply:
fixed